go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy_controller.go

```diff
@@ -431,25 +436,29 @@ func (c *Controller) onANPPodUpdate(oldObj, newObj interface{}) {
 		return
 	}
 	// We only care about pod's label changes, pod's IP changes
-	// and pod going into completed state. Rest of the cases we may return
+	// pod going into completed state and pod getting scheduled and switching
+	// zones. Rest of the cases we may return
 	oldPodLabels := labels.Set(oldPod.Labels)
 	newPodLabels := labels.Set(newPod.Labels)
 	oldPodIPs, _ := util.GetPodIPsOfNetwork(oldPod, &util.DefaultNetInfo{})
 	newPodIPs, _ := util.GetPodIPsOfNetwork(newPod, &util.DefaultNetInfo{})
 	oldPodCompleted := util.PodCompleted(oldPod)
 	newPodCompleted := util.PodCompleted(newPod)
+	oldPodLocal := c.isPodInLocalZone(oldPod)
+	newPodLocal := c.isPodInLocalZone(newPod)
```

Comment:
yup, pod cannot be re-scheduled from one node to another unless we live migrated it which is a kubevirt only case.

the reason we have this check above is first when the pod is not scheduled it is considered as remote pod, then when it get's scheduled we get an update since pod.Spec.NodeName is set, then we know pod went from remote->local and we take action to do its setup if it matches the subject of the ANP.

pod going from local->remote can only happen during upgrades but that means I will just delete the pod's UUID from my PG however that will also mean pod controller will just remove the pod's LSP.

Reply:
> the reason we have this check above is first when the pod is not scheduled it is considered as remote pod, then when it get's scheduled we get an update since pod.Spec.NodeName is set, then we know pod went from remote->local and we take action to do its setup if it matches the subject of the ANP.

OK but this is basically a check for pod scheduling, regardless of zone. I don't see how the zones have anything to do with it.

> pod going from local->remote can only happen during upgrades but that means I will just delete the pod's UUID from my PG however that will also mean pod controller will just remove the pod's LSP.

But why would we need to be aware of this in a pod update event? This is either noticed after a restart, for which we would have a pod add event, or through a node update event, which we are not handling at all in this controller.

Reply:
> > the reason we have this check above is first when the pod is not scheduled it is considered as remote pod, then when it get's scheduled we get an update since pod.Spec.NodeName is set, then we know pod went from remote->local and we take action to do its setup if it matches the subject of the ANP.
> 
> OK but this is basically a check for pod scheduling, regardless of zone. I don't see how the zones have anything to do with it.

but this event for when a pod is scheduled is becoming important for us only because of IC and zones, so while its a check for pod scheduling i never had to do it for nonIC because all pods are automatically local to me, we would anyways get an event for pod add like you said, this can be changed to `isPodScheduled` if it makes it less confusing, but I'd prefer we keep this as is, I can leave some comments to clarify that this is indeed for pod scheduling changes and nothing else.

> 
> > pod going from local->remote can only happen during upgrades but that means I will just delete the pod's UUID from my PG however that will also mean pod controller will just remove the pod's LSP.
> 
> But why would we need to be aware of this in a pod update event? This is either noticed after a restart, for which we would have a pod add event, or through a node update event, which we are not handling at all in this controller.

ack i agree with this. ^ the logic today solely exists for when pod scheduled is jumping from remote->local.

Reply:
I have written detailed comments for this one. I hope that will work. leaving this unresolved till you read the new comments and are ok with it, if you feel strongly that we should change it to `IsPodScheduled` instead, then we could do that

Reply:
Do we actually need the check? Wouldn't the pod IPs check be enough? What doesn't work if we do not have this check in place?

I won't press it as long as it is clear this is not meant to support any kind of zone migration. The check is not intuitive for me, but it might be me.

Reply:
yes the podIPs check is enough for 99% of the cases probably because we are sure we will get a podIP update, its just to make sure this scenario is thought of... Probably removing this is totally ok...
ok i am going to remove this check from podupdate, since this feature should always get the podIP/podStatus updates